src/Language/PureScript/Sugar/Names/Imports.hs

```diff
@@ -162,11 +163,13 @@ resolveModuleImport
    . (Applicative m, MonadError MultipleErrors m, MonadWriter MultipleErrors m)
   => Env
   -> Imports
-  -> (ModuleName, [(Maybe SourceSpan, ImportDeclarationType, Maybe ModuleName)])
+  -> (ModuleName, [(Maybe SourceSpan, Maybe ImportDeclarationType, Maybe ModuleName)])
```

Comment:
When `Nothing` it's the current ("self") module temporarily injected into the environment to make local names resolvable.

Later on the `ImportDeclarationType` gets mapped to `ImportProvenance`, and `Nothing -> Local`. I didn't want to add a value to `ImportDeclarationType` that only exists for a moment during name desugaring.